DNN Platform/DotNetNuke.Abstractions/Logging/IEventLogger.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.Abstractions.Logging
+{
+    using System.Collections;
+    using System.Collections.Generic;
+
+    using DotNetNuke.Abstractions.Portals;
+
+    /// <summary>
+    /// Event Logger.
+    /// </summary>
+    public interface IEventLogger
```

Comment:
My only other feedback is wondering whether `IEventLogger` should be primarily focused on `AddLog` and related methods, and the log config APIs could be put into a separate interface (later).

Reply:
Thanks for all the feedback here, it is much appreciated. I am glad you commented on this as it gives me an opportunity to explain my thoughts.

## My Reasoning behind IEventLogger
I decided to make this PR a lift and shift from `IEventLogController` to `IEventLogger`. I think it makes sense to have 2 different types of logging interfaces: log4net file logging vs database event logging. The event logging appears to be designed around tracking business rule events that happen from different modules or core components. Where log4net is more for tracking exceptions and errors. Ultimately my goal would be to submit a PR that removes log4net and uses `Microsoft.Extensions.Logging` that would use the `ILogger<T>` that ships with that NuGet. I just don't think we are at the point where we can do that yet.

I have quite a bit of experience with event tracking as it is very popular in mobile and desktop apps. The problem I see most of the time is an `ILogger` API structure doesn't line up very well with something like the `IEventLogger` API structure. I think it makes sense to keep them separate. Once we get better support for logging in DNN with Dependency Injection I think it would make sense to add an AggregateLogger or extension methods so you can use general Logging and Event Logging from the same interface. I just don't think we are there, hence the lift and shift PR here.

## Future Plans
I am not a big fan of how hard it is to get up and running with the `EventLogController` and now the `IEventLogger`. There is a lot of guess and check as the database doesn't like null values. I would like to simplify this API structure if possible, but I think the goal of this PR would be to just do a lift and shift since it just needs some minor improvements.

For example I think it would make more sense to use `IDictionary<string, string>` for `LogProperties` instead of `ArrayList`. I'm sure this was first implemented at a time when generics weren't available so this was the best way to handle it at the time.

If we accept this lift and shift PR I plan to create a new backlog item where we can discuss changes to the API structure

Reply:
To directly answer your question, I agree we should move non-logging methods to a separate interface. Any thoughts on naming?

Perhaps we call it the `IEventLogManager` or `IEventLogService`. The challenge with a separate interface will be too many interfaces with similar names as it could cause confusion.

Reply:
How about `IEventLogConfigService`?

Reply:
I just want to say a few things here....

1.  Totally support this, but I think we non-logging stuff to `IEventLogConfigManager` as I think that follows our new convention?  (Or would _Service be more appropriate.)
2. I'm 1000000% on board with `ILogger<T>` support!  As ultimately I want to be able to add SeriLog

Reply:
I have added an `IEventLogConfigService` in the latest code push. I am going to resolve this thread so we can focus the conversation on individual APIs in that interface